File: Lib/Models/Media.cs
using System.Diagnostics.CodeAnalysis;

namespace Lib.Models;

[ExcludeFromCodeCoverage]
public class Media : BaseModel
{
    public Guid MediaId { get; set; } = Guid.NewGuid();
    public string OriginalFilename { get; set; }
    public string Extension { get; set; }
    public string MimeType { get; set; }
    public long Size { get; set; }
    public string StoragePath { get; set; }
    public string Hash { get; set; }
    public string Metadata { get; set; }
}

File: App/ClientApp/src/app/shared/models/media.ts
export interface Media {
  mediaId: string;
  originalFilename: string;
  extension: string;
  mimeType: string;
  size: number;
  storagePath: string;
  hash: string;
  metadata: string;
  // BaseModel properties
  createdAt: Date;
  modifiedAt: Date;
  userId: string;
  concurrencyStamp: string;
}

export const defaultMedia: Media = {
  mediaId: "",
  originalFilename: "",
  extension: "",
  mimeType: "",
  size: 0,
  storagePath: "",
  hash: "",
  metadata: "",
  createdAt: new Date(),
  modifiedAt: new Date(),
  userId: "",
  concurrencyStamp: "",
};
